src/libsystemd-network/network-internal.c

```diff
@@ -509,3 +509,31 @@ int deserialize_dhcp_routes(struct sd_dhcp_route **ret, size_t *ret_size, size_t
 
         return 0;
 }
+
+int serialize_dhcp_option(FILE *f, const char *key, const uint8_t *data, size_t size) {
+        _cleanup_free_ char *hex_buf = NULL;
+
+        assert(f);
+        assert(key);
+        assert(data);
+        assert(size);
```

Comment:
By convention, in systemd integers are usually not used directly as booleans, but are instead compared to 0. So this should be assert(size > 0).

But a bigger question is why you forbid size == 0. Shouldn't this be allowed?